google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddShoppingSmartAd.java

```diff
@@ -0,0 +1,398 @@
+import com.google.ads.googleads.v0.enums.AdGroupStatusEnum.AdGroupStatus;
+import com.google.ads.googleads.v0.enums.AdGroupTypeEnum.AdGroupType;
+import com.google.ads.googleads.v0.enums.AdvertisingChannelSubTypeEnum.AdvertisingChannelSubType;
+import com.google.ads.googleads.v0.enums.AdvertisingChannelTypeEnum.AdvertisingChannelType;
+import com.google.ads.googleads.v0.enums.BudgetDeliveryMethodEnum.BudgetDeliveryMethod;
+import com.google.ads.googleads.v0.enums.CampaignStatusEnum.CampaignStatus;
+import com.google.ads.googleads.v0.enums.ListingGroupTypeEnum.ListingGroupType;
+import com.google.ads.googleads.v0.errors.GoogleAdsError;
+import com.google.ads.googleads.v0.resources.Ad;
+import com.google.ads.googleads.v0.resources.AdGroup;
+import com.google.ads.googleads.v0.resources.AdGroupAd;
+import com.google.ads.googleads.v0.resources.AdGroupCriterion;
+import com.google.ads.googleads.v0.resources.Campaign;
+import com.google.ads.googleads.v0.resources.Campaign.ShoppingSetting;
+import com.google.ads.googleads.v0.resources.CampaignBudget;
+import com.google.ads.googleads.v0.services.AdGroupAdOperation;
+import com.google.ads.googleads.v0.services.AdGroupAdServiceClient;
+import com.google.ads.googleads.v0.services.AdGroupCriterionOperation;
+import com.google.ads.googleads.v0.services.AdGroupCriterionServiceClient;
+import com.google.ads.googleads.v0.services.AdGroupOperation;
+import com.google.ads.googleads.v0.services.AdGroupServiceClient;
+import com.google.ads.googleads.v0.services.CampaignBudgetOperation;
+import com.google.ads.googleads.v0.services.CampaignBudgetServiceClient;
+import com.google.ads.googleads.v0.services.CampaignOperation;
+import com.google.ads.googleads.v0.services.CampaignServiceClient;
+import com.google.ads.googleads.v0.services.MutateAdGroupAdResult;
+import com.google.ads.googleads.v0.services.MutateAdGroupCriterionResult;
+import com.google.ads.googleads.v0.services.MutateAdGroupResult;
+import com.google.ads.googleads.v0.services.MutateCampaignBudgetsResponse;
+import com.google.ads.googleads.v0.services.MutateCampaignResult;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.BoolValue;
+import com.google.protobuf.DoubleValue;
+import com.google.protobuf.Int64Value;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Collections;
+
+/**
+ * This example creates a Smart Shopping campaign, a Smart Shopping ad group, a Smart Shopping ad
+ * group ad and a listing group for "All products".
+ *
+ * <p>Prerequisite: You need to have access to a Merchant Center account. You can find instructions
+ * to create a Merchant Center account here: https://support.google.com/merchants/answer/188924.
+ * This account must be linked to your AdWords account. The integration instructions can be found
+ * at: https://developers.google.com/adwords/shopping/full-automation/articles/t15.
+ */
+public class AddShoppingSmartAd {
```

Comment:
PTAL at the presubmit failure as well. Looks like poms may be incorrect. Error is:

    Non-resolvable parent POM for com.google.api-ads:google-ads-examples:0.4.1-SNAPSHOT:
    Could not find artifact com.google.api-ads:google-ads-parent:pom:0.4.1-SNAPSHOT
    and 'parent.relativePath' points at wrong local POM @ line 22, column 11

Reply:
All sorted, thanks! Local branch was a few commits behind master.